Add Check Support for Java 21 Record Pattern Syntax: OperatorWrap

child of https://github.com/checkstyle/checkstyle/issues/14890
I have read check documentation: https://checkstyle.org/checks/whitespace/operatorwrap.html#OperatorWrap
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test2.java                                                
record ColoredPoint(int p, int x, String c) { }
record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }
public class Test2 {
    void test(Object obj) {
        if (!(obj instanceof                        // violation
                Rectangle(ColoredPoint record,ColoredPoint yield))) { 
        }
    }

}
PS D:\CS\test> cat config.xml                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="OperatorWrap">

        </module>

    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test2.java
Starting audit...
[ERROR] D:\CS\test\src\Test2.java:5:19: 'instanceof' should be on a new line. [OperatorWrap]
Audit done.
Checkstyle ends with 1 errors.
PS D:\CS\test> 

```
Describe what you expect in detail.

This can be closed without any updates

--- a/config/jsoref-spellchecker/whitelist.words
+++ b/config/jsoref-spellchecker/whitelist.words
@@ -967,6 +967,7 @@ opencollective
 openjdk
 openstreetmap
 operatorwrap
+opsummary
 optgroup
 optimisation
 optionalbracesusage

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
@@ -31,7 +31,13 @@ import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 
 /**
  * <p>
- * Checks the policy on how to wrap lines on operators.
+ * Checks the policy on how to wrap lines on
+ * <a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/opsummary.html">
+ * operators</a>.
+ * </p>
+ * <p>
+ * See the <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.20.2">
+ * Java Language Specification</a> for more information about {@code instanceof} operator.
  * </p>
  * <ul>
  * <li>

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/OperatorWrapCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/OperatorWrapCheck.xml
@@ -5,7 +5,13 @@
              name="OperatorWrap"
              parent="com.puppycrawl.tools.checkstyle.TreeWalker">
          <description>&lt;p&gt;
- Checks the policy on how to wrap lines on operators.
+ Checks the policy on how to wrap lines on
+ &lt;a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/opsummary.html"&gt;
+ operators&lt;/a&gt;.
+ &lt;/p&gt;
+ &lt;p&gt;
+ See the &lt;a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.20.2"&gt;
+ Java Language Specification&lt;/a&gt; for more information about {@code instanceof} operator.
  &lt;/p&gt;</description>
          <properties>
             <property default-value="nl"

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheckTest.java
@@ -192,4 +192,35 @@ public class OperatorWrapCheckTest
         verifyWithInlineConfigParser(
                 getPath("InputOperatorWrapWithTrimOptionProperty.java"), expected);
     }
+
+    @Test
+    public void testInstanceOfOperator() throws Exception {
+        final String[] expected = {
+            "17:15: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "23:15: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "35:23: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "39:23: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "49:33: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "59:33: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "72:15: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+            "82:38: " + getCheckMessage(MSG_LINE_NEW, "instanceof"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputOperatorWrapInstanceOfOperator.java"), expected);
+    }
+
+    @Test
+    public void testInstanceOfOperatorEndOfLine() throws Exception {
+        final String[] expected = {
+            "28:17: " + getCheckMessage(MSG_LINE_PREVIOUS, "instanceof"),
+            "43:17: " + getCheckMessage(MSG_LINE_PREVIOUS, "instanceof"),
+            "65:20: " + getCheckMessage(MSG_LINE_PREVIOUS, "instanceof"),
+            "78:17: " + getCheckMessage(MSG_LINE_PREVIOUS, "instanceof"),
+            "88:21: " + getCheckMessage(MSG_LINE_PREVIOUS, "instanceof"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                        "InputOperatorWrapInstanceOfOperatorEndOfLine.java"), expected);
+    }
+
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapInstanceOfOperator.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapInstanceOfOperator.java
@@ -1,0 +1,93 @@
+/*
+OperatorWrap
+option = (default)nl
+tokens = (default)QUESTION, COLON, EQUAL, NOT_EQUAL, DIV, PLUS, MINUS, STAR, MOD, \
+         SR, BSR, GE, GT, SL, LE, LT, BXOR, BOR, LOR, BAND, LAND, TYPE_EXTENSION_AND, \
+         LITERAL_INSTANCEOF
+
+
+*/
+//non-compiled with javac: Compilable with Java21
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.operatorwrap;
+
+public class InputOperatorWrapInstanceOfOperator {
+    void test(Object o) {
+        // violation below, ''instanceof' should be on a new line.'
+        if (o instanceof
+                Integer) {}
+
+        if (o instanceof Integer) {}
+
+        // violation below, ''instanceof' should be on a new line.'
+        if (o instanceof
+                Integer i) {}
+
+        if (o instanceof Integer i) {}
+
+        if (o
+                instanceof Integer) {
+        }
+    }
+
+    void test2(Object o) {
+        // violation below, ''instanceof' should be on a new line.'
+        boolean a = o instanceof
+                Integer;
+        boolean b = o instanceof Integer;
+        // violation below, ''instanceof' should be on a new line.'
+        boolean c = o instanceof
+                Integer i;
+        boolean d = o instanceof Integer i;
+        boolean e = o
+                instanceof Integer i;
+    }
+
+    void test3(Object o) {
+       switch (o) {
+           // violation below, ''instanceof' should be on a new line.'
+           case Number n when n instanceof
+                   Integer: { } break;
+           default: { }
+       }
+       switch (o) {
+           case Number n when n instanceof Integer: { } break;
+           default: { }
+       }
+       switch (o) {
+           // violation below, ''instanceof' should be on a new line.'
+           case Number i when i instanceof
+                   Integer _ : { } break;
+           default: { }
+       }
+        switch (o) {
+           case Number n when n
+                   instanceof Integer: { } break;
+           default: { }
+       }
+    }
+
+    void test4(Object o) {
+        // violation below, ''instanceof' should be on a new line.'
+        if (o instanceof
+                Point(int x, int y)) {}
+
+        if (o instanceof Point(int x, int y)) {}
+
+        if (o
+                instanceof Point(int x, int y)) {}
+
+        switch (o) {
+            // violation below, ''instanceof' should be on a new line.'
+            case Object obj when obj instanceof
+                    Point(int _, int _): {} break;
+            default: {}
+        }
+        switch (o) {
+            case Object obj when obj
+                    instanceof Point(int _, int _): {} break;
+            default: {}
+        }
+    }
+    record Point(int x,int y) {}
+}

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapInstanceOfOperatorEndOfLine.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapInstanceOfOperatorEndOfLine.java
@@ -1,0 +1,93 @@
+/*
+OperatorWrap
+option = eol
+tokens = (default)QUESTION, COLON, EQUAL, NOT_EQUAL, DIV, PLUS, MINUS, STAR, MOD, \
+         SR, BSR, GE, GT, SL, LE, LT, BXOR, BOR, LOR, BAND, LAND, TYPE_EXTENSION_AND, \
+         LITERAL_INSTANCEOF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.whitespace.operatorwrap;
+
+public class InputOperatorWrapInstanceOfOperatorEndOfLine {
+    void test(Object o) {
+        if (o instanceof
+                Integer) {}
+
+        if (o instanceof Integer) {}
+
+        if (o instanceof
+                Integer i) {}
+
+        if (o instanceof Integer i) {}
+
+        if (o
+                // violation below, ''instanceof' should be on the previous line.'
+                instanceof Integer) {
+        }
+    }
+
+    void test2(Object o) {
+
+        boolean a = o instanceof
+                Integer;
+        boolean b = o instanceof Integer;
+
+        boolean c = o instanceof
+                Integer i;
+        boolean d = o instanceof Integer i;
+        boolean e = o
+                // violation below, ''instanceof' should be on the previous line.'
+                instanceof Integer i;
+    }
+
+    void test3(Object o) {
+       switch (o) {
+           case Number n when n instanceof
+                   Integer: { } break;
+           default: { }
+       }
+       switch (o) {
+           case Number n when n instanceof Integer: { } break;
+           default: { }
+       }
+       switch (o) {
+
+           case Number i when i instanceof
+                   Integer _ : { } break;
+           default: { }
+       }
+        switch (o) {
+           case Number n when n
+                  // violation below, ''instanceof' should be on the previous line.'
+                   instanceof Integer: { } break;
+           default: { }
+       }
+    }
+
+    void test4(Object o) {
+        if (o instanceof
+                Point(int x, int y)) {}
+
+        if (o instanceof Point(int x, int y)) {}
+
+        if (o
+                // violation below, ''instanceof' should be on the previous line.'
+                instanceof Point(int x, int y)) {}
+
+        switch (o) {
+            case Object obj when obj instanceof
+                    Point(int _, int _): {} break;
+            default: {}
+        }
+        switch (o) {
+            case Object obj when obj
+                   // violation below, ''instanceof' should be on the previous line.'
+                    instanceof Point(int _, int _): {} break;
+            default: {}
+        }
+    }
+    record Point(int x,int y) {}
+}

--- a/src/xdocs/checks/whitespace/operatorwrap.xml
+++ b/src/xdocs/checks/whitespace/operatorwrap.xml
@@ -10,7 +10,13 @@
       <p>Since Checkstyle 3.0</p>
       <subsection name="Description" id="Description">
         <p>
-          Checks the policy on how to wrap lines on operators.
+          Checks the policy on how to wrap lines on
+          <a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/opsummary.html">
+          operators</a>.
+        </p>
+        <p>
+          See the <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.20.2">
+          Java Language Specification</a> for more information about {@code instanceof} operator.
         </p>
       </subsection>
 

--- a/src/xdocs/checks/whitespace/operatorwrap.xml.template
+++ b/src/xdocs/checks/whitespace/operatorwrap.xml.template
@@ -10,7 +10,13 @@
       <p>Since Checkstyle 3.0</p>
       <subsection name="Description" id="Description">
         <p>
-          Checks the policy on how to wrap lines on operators.
+          Checks the policy on how to wrap lines on
+          <a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/opsummary.html">
+          operators</a>.
+        </p>
+        <p>
+          See the <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.20.2">
+          Java Language Specification</a> for more information about {@code instanceof} operator.
         </p>
       </subsection>
 

